DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionTask+Tracking.swift

```diff
@@ -0,0 +1,28 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+private var sessionFirstPartyHostsKey: UInt8 = 31
+internal extension URLSessionTask {
+    /// Returns the first party hosts for this task.
+    var firstPartyHosts: FirstPartyHosts {
+        get {
+            return sessionFirstPartyHosts ?? .init()
+        }
+    }
+
+    /// Extension property for storing first party hosts passed from `URLSession` to `URLSessionTask`.
+    /// This is used for `URLSessionTask` based APIs.
+    var sessionFirstPartyHosts: FirstPartyHosts? {
+        set {
+            objc_setAssociatedObject(self, &sessionFirstPartyHostsKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
question/ Is non-atomic on purpose (vs atomic [`OBJC_ASSOCIATION_RETAIN`](https://developer.apple.com/documentation/objectivec/objc_associationpolicy/objc_association_retain))?

Reply:
Initially, I didn't see a reason for it to be atomic but now for both `URLSessionDelegate.core` and `URLSessionTask.sessionFirstPartyHosts`, it makes sense to be atomic because we are not sure how these properties are called and on which thread depending on the delegate method execution and URLSession task method execution.

Good call.